packages/ui/components/ui/dialog.tsx

```diff
@@ -28,66 +29,83 @@ const DialogOverlay = React.forwardRef<
+    'data-[state=closed]:fade-out-0',
+    'data-[state=open]:fade-in-0',
+    'data-[state=closed]:zoom-out-95',
+    'data-[state=open]:zoom-in-95',
+    'data-[state=closed]:slide-out-to-left-1/2',
+    'data-[state=closed]:slide-out-to-top-[48%]',
+    'data-[state=open]:slide-in-from-left-1/2',
+    'data-[state=open]:slide-in-from-top-[48%]',
+    'sm:rounded-lg',
+    'data-[state=closed]:fill-mode-forwards',
+    'data-[state=open]:fill-mode-forwards',
+  ],
+  {
+    variants: {
+      size: {
+        lg: ['max-w-5xl'],
+        sm: ['max-w-[312px]', 'md:max-w-[400px]'],
+      },
+    },
+    defaultVariants: {
+      size: 'sm',
+    },
+  },
+);
+interface DialogContentProps extends VariantProps<typeof dialogContentVariants> {
+  children?: React.ReactNode;
+}
+
 const DialogContent = React.forwardRef<
   React.ElementRef<typeof DialogPrimitive.Content>,
-  React.ComponentPropsWithoutRef<typeof DialogPrimitive.Content>
->(({ className, children, ...props }, ref) => (
+  DialogContentProps
+>(({ children, size }, ref) => (
   <DialogPortal>
     <DialogOverlay />
-    <DialogPrimitive.Content
-      ref={ref}
-      className={cn(
-        'fixed',
-        'left-[50%]',
-        'top-[50%]',
-        'z-50',
-        'grid',
-        'w-full',
-        'max-w-lg',
-        'max-h-screen',
-        'translate-x-[-50%]',
-        'translate-y-[-50%]',
-        'gap-4',
-        'rounded-lg',
-        'bg-background',
-        'shadow-lg',
-        'duration-200',
-        'data-[state=open]:animate-in',
-        'data-[state=closed]:animate-out',
-        'data-[state=closed]:fade-out-0',
-        'data-[state=open]:fade-in-0',
-        'data-[state=closed]:zoom-out-95',
-        'data-[state=open]:zoom-in-95',
-        'data-[state=closed]:slide-out-to-left-1/2',
-        'data-[state=closed]:slide-out-to-top-[48%]',
-        'data-[state=open]:slide-in-from-left-1/2',
-        'data-[state=open]:slide-in-from-top-[48%]',
-        'sm:rounded-lg',
-        'md:w-full',
-        'data-[state=closed]:fill-mode-forwards',
-        'data-[state=open]:fill-mode-forwards',
-        className,
-      )}
-      {...props}
-    >
+    <DialogPrimitive.Content ref={ref} className={cn(dialogContentVariants({ size }))}>
       {children}
-      <DialogPrimitive.Close className='absolute left-[30px] top-[23px] z-10 rounded-sm text-muted-foreground ring-offset-background transition-opacity hover:opacity-50 focus:outline-none focus:ring-0 focus:ring-offset-2 disabled:pointer-events-none data-[state=open]:bg-accent'>
-        <Cross2Icon className='size-6' />
-        <span className='sr-only'>Close</span>
-      </DialogPrimitive.Close>
     </DialogPrimitive.Content>
   </DialogPortal>
 ));
 DialogContent.displayName = DialogPrimitive.Content.displayName;
 
-const DialogHeader = ({ className, ...props }: React.HTMLAttributes<HTMLDivElement>) => (
+const DialogHeader = ({ children }: { children?: React.ReactNode }) => (
```

Comment:
No longer configurable via `className`, to ensure consistent appearance. Also, the header now includes the close button, rather than having an absolute-positioned close button in `<DialogContent />`.